src/clj/rems/standalone.clj

```diff
@@ -116,102 +116,108 @@
+
+          "api-key"
+          (let [[_ command api-key & command-args] args]
+            (mount/start #'rems.config/env #'rems.db.core/*db*)
+            (case command
+              "get" (do)
+              "add" (api-key/update-api-key! api-key {:comment (str/join " " command-args)})
+              "delete" (api-key/delete-api-key! api-key)
+              "set-users" (api-key/update-api-key! api-key {:users command-args})
+              "allow" (let [[method path] command-args
+                            entry {:method method :path path}
+                            old (:paths (api-key/get-api-key api-key))]
+                        (api-key/update-api-key! api-key {:paths (conj old entry)}))
+              "allow-all" (api-key/update-api-key! api-key {:paths nil})
+              (do (usage)
+                  (System/exit 1)))
+            (if api-key
+              (prn (api-key/get-api-key api-key))
+              (mapv prn (api-key/get-api-keys))))
+
+          "ega"
+          (let [[_ command userid username password config-id & _] args]
+            (mount/start #'rems.config/env #'rems.db.core/*db*)
+            (case command
+              "api-key" (let [ega-config (->> (:entitlement-push env)
+                                              (filter (comp #{:ega} :type))
+                                              (find-first (comp #{config-id} :id)))]
+                          (assert ega-config (str "Could not find :entitlement-push with :type :ega and :id " (pr-str config-id)))
+                          (ega/generate-api-key-with-account {:userid userid :username username :password password :config ega-config}))
+              (do (usage)
+                  (System/exit 1))))
+
+          "list-users"
+          (do
+            (mount/start #'rems.config/env #'rems.db.core/*db*)
+            (doseq [u (users/get-all-users)]
+              (-> u
+                  (assoc :roles (roles/get-roles (:userid u)))
+                  json/generate-string
+                  println)))
+
+          "grant-role"
+          (let [[_ role user] args]
+            (if (not (and role user))
+              (do (usage)
+                  (System/exit 1))
+              (do (mount/start #'rems.config/env #'rems.db.core/*db*)
+                  (roles/add-role! user (keyword role)))))
+
+          "remove-role"
+          (let [[_ role user] args]
+            (if (not (and role user))
+              (do (usage)
+                  (System/exit 1))
+              (do (mount/start #'rems.config/env #'rems.db.core/*db*)
+                  (roles/remove-role! user (keyword role)))))
+
+          "validate"
+          (do
+            (mount/start #'rems.config/env #'rems.db.core/*db*)
+            (when-not (validate/validate)
+              (System/exit 2)))
+
+          ;; show usage if command is unrecognized
+          (usage))
+        ;; exit CLI after succesful command execution
+        (System/exit 0)))))
```

Comment:
the CLI app is left running indefinitely normally which I thought is a little peculiar. is there a use case that it should "stand by" after running the issued command?

Reply:
Definitely we can have it exit and it is better like that. This is likely just leftover behavior from years ago. Like ideally we would probably want to have a separate `start` command to start the server and get the usage from the default, but the default behavior to start the server can stay as it is backwards compatible. 

Btw. looks like the other commands mostly exit with code `1` after printing `(usage)`.

Reply:
exiting with code `1` is correct in those cases, as they seem to be an error scenario. in general i think the command should define the exit code and otherwise interpret that everything went according to plan. now that i think of it, the last catch-all `(usage)` should also exit with 1 to make it clear it's an error scenario.